lib/active_merchant/billing/gateways/pin.rb

```diff
@@ -149,23 +149,56 @@ def add_capture(post, options)
         post[:capture] = capture != false
       end
 
-      def add_creditcard(post, creditcard)
-        if creditcard.respond_to?(:number)
-          post[:card] ||= {}
+      def add_payment_method(post, payment_method)
+        return unless payment_method
 
+        case payment_method
+        when NetworkTokenizationCreditCard
+          token = get_single_use_token(payment_method, options)
```

Comment:
💬 Comment/Style:

If the `token` variable is just a 'transition' variable, maybe a direct assignment to the post hash is better:

```ruby
   post[:payment_source_token] = get_single_use_token(payment_method, options)
```